lib/dor/workflow/client/status.rb

```diff
@@ -39,43 +39,12 @@ def initialize(druid:, version:, lifecycle_routes:)
           @druid = druid
           @version = version.to_s
           @lifecycle_routes = lifecycle_routes
-        end
-
-        # @return [Hash{Symbol => Object}] including :status_code and :status_time
-        def info
-          @info ||= begin
-            accessioned_milestones = current_milestones.select { |m| m[:milestone] == 'accessioned' }
-            if accessioned_milestones.any?
-              # if we have an accessioned milestone, this is the last possible step and should be the status regardless of timestamp
-              { status_code: STEPS['accessioned'], status_time: accessioned_milestones.last[:at].utc.xmlschema }
-            else
-              status_code = 0
-              status_time = nil
-              # for each milestone in the current version, see if it comes at the same time or after the current 'last' step.
-              # if so, make it the last and record the date/time
-              current_milestones.each do |m|
```

Comment:
And this is another instance of `#each` as an intention-befuddling method. Here the `#each` is acting as both a filter (removing milestones without a STEPS-matching name) and a max-value finder. But Ruby already has iterator methods that can do both of these operations.